File: pinyougou-user-web/src/main/webapp/js/service/addressService.js
//服务层
app.service('addressService',function($http){
	    	
	//查询实体
	this.findOne=function(id){
		return $http.get('../address/findOne.do?id='+id);
	}
	
	
	//删除
	this.dele=function(address){
		return $http.post('address/delete.do',address);
	}
   	
	//查询收货地址列表
	this.findAddressList=function(){
		return $http.get("address/findListByLoginUser.do");
	}
	
	//新增收货地址
	this.add=function(address){
		return $http.post("address/addAddress.do",address);
	}
	//修改 
	this.update=function(address){
		return  $http.post('address/update.do',address );
	}
	//设为默认地址
	this.setToDeafult=function(address){
		return $http.post("address/setToDeafult.do",address);
	}
	
	//查询省份 列表
	this.findProvince=function(){
		return $http.get("address/findProvince.do");
	}
	//查询市级 列表
	this.findCity=function(provinceId){
		return $http.get("address/findCity.do?provinceId="+provinceId);
	}
	//查询县级 列表
	this.findarea=function(cityid){
		return $http.get("address/findarea.do?cityid="+cityid);
	}
});

File: pinyougou-user-interface/src/main/java/com/pinyougou/user/service/AddressService.java
package com.pinyougou.user.service;
import java.util.List;
import com.pinyougou.pojo.TbAddress;
import com.pinyougou.pojo.TbAreas;
import com.pinyougou.pojo.TbCities;
import com.pinyougou.pojo.TbProvinces;

import entity.PageResult;
/**
 * 服务层接口
 * @author Administrator
 *
 */
public interface AddressService {

	/**
	 * 返回全部列表
	 * @return
	 */
	public List<TbAddress> findAll();
	
	
	/**
	 * 返回分页列表
	 * @return
	 */
	public PageResult findPage(int pageNum,int pageSize);
	
	
	/**
	 * 增加
	*/
	public void add(TbAddress address);
	
	
	/**
	 * 修改
	 */
	public void update(TbAddress address);
	

	/**
	 * 根据ID获取实体
	 * @param id
	 * @return
	 */
	public TbAddress findOne(Long id);
	
	
	/**
	 * 批量删除
	 * @param ids
	 */
	public void delete(Long id);

	/**
	 * 分页
	 * @param pageNum 当前页 码
	 * @param pageSize 每页记录数
	 * @return
	 */
	public PageResult findPage(TbAddress address, int pageNum,int pageSize);
	
	/**
	 * 根据用户名查询地址
	 * @param userId
	 * @return
	 */
	public List<TbAddress> findListByUserId(String userId);
	
	
	/**
	 * 修改默认地址
	 * @param address
	 */
	public void setToDeafult(TbAddress address);



	//查询省/市/区列表
	public List<TbProvinces> findProvince();
	public List<TbCities> findCity(String provinceId);
	public List<TbAreas> findarea(String cityid);
}
